src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/VariantEval.java

```diff
@@ -466,66 +361,41 @@ protected EvaluationContext createEvaluationContext(final Set<Class<? extends Va
         return new EvaluationContext(this, evaluationObjects);
     }
 
-    private class PositionAggregator {
-        private SimpleInterval i = null;
-
-        private ReadsContext readsContext;
-        private ReferenceContext referenceContext;
-        private FeatureContext featureContext;
-
-        private void addVariant(VariantContext vc, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
-            if (i == null || !vc.getContig().equals(i.getContig()) || vc.getStart() != i.getStart()) {
-                callDoApply();
+    public String getNameForInput(FeatureInput<VariantContext> input) {
+        return inputToNameMap.get(input);
+    }
 
-                i = new SimpleInterval(vc.getContig(), vc.getStart(), vc.getEnd());
-                this.readsContext = readsContext;
-                this.referenceContext = referenceContext;
-                this.featureContext = featureContext;
-            }
-            else if (vc.getEnd() > i.getEnd()) {
-                //expand region
-                i = new SimpleInterval(i.getContig(), i.getStart(), vc.getEnd());
+    @Override
+    public void apply(List<VariantContext> variantContexts, ReferenceContext referenceContext, List<ReadsContext> readsContexts) {
+        Map<FeatureInput<VariantContext>, List<VariantContext>> variantMap = groupVariantsByFeatureInput(variantContexts);
```

Comment:
OK. for the record, other subclasses of MultiVariantWalkerGroupedOnStart, like FilterAlignmentArtifacts and CombineGVCFs dont have final either. I assume everyone just let Intellij auto-generate the method, so it would inherit the argument pattern from MultiVariantWalkerGroupedOnStart...